### ISSUE TITLE
qTox should use file manager from system

### ISSUE THREAD

**Issue by ovalseven8** [qTox/qTox#2499]
Hello,

I'm not sure if there is already an issue about that, but I couldn't find one.

At the moment, qTox isn't using the file manager from the system (for example PCManFM when using Lubuntu). Instead, there is, well I don't know, this one: ![Screenshot](https://cloud.githubusercontent.com/assets/8258609/10971742/116eff1c-83d6-11e5-8a7d-b44433f7cbcb.png)

**Comment by agilob** [comment]
it is one provided by Qt, it's multiplatform and more flexible. there was an issue about it already somewhere.

**Comment by ProMcTagonist** [comment]
> multiplatform

like java

> more flexible. 
- no Recent
- starts in an inappropriate place
- awkward to navigate
- no media previews
- looks awful

What does "flexible" mean?

**Comment by agilob** [comment]
> like java

 dont bother supporting different types of file pickers with different implementations which is not standarised

> awkward to navigate

there are 5 different kinds of file pickers in Qt, feel free to investigate them and see how they look on different platforms, this one is universal

> no Recent

d&d to chat area, only times when you **need** file picker is to import profile, export profile and choose autoaccept directory - which are directories not files so "recent" doesn't help

> starts in an inappropriate place

?

> looks awful

there are 5 different kinds of file pickers in Qt, feel free to investigate them and see how they look on different platforms, this one is universal

We already bothered to support a few types of system trays, remember how we ended up?

**Comment by zetok** [comment]
@ovalseven8 this seems to me like an upstream problem - i.e. Qt will try to use something that is a better file picker in your system, but if it won't find one, it'll default to this one.

I.e. it's a problem with the distro that you use, since it doesn't support file picker acceptable by Qt5.

With that being said, note that I on Gentoo also have this ugly one.

On the other hand, e.g. Arch users who use KDE5 have qTox using the KDE5 file picker. ← if that's not the case, then this is a bug, otherwise, there isn't much that can be done.

**Comment by ovalseven8** [comment]
@zetok: Well, I just tried with Kubuntu 15.10 (Plasma 5) and there wasn't the default file manager too. I think qTox should be able to find a better solution.

@tux3: Do you have any idea how to fix that problem?

I'm not sure if it's helpful:
- https://forum.qt.io/topic/17087/how-to-open-a-folder-using-default-file-explorer-or-any-file-using-the-default-application-for-that-file
- https://stackoverflow.com/questions/10250332/qt-how-to-open-a-link-in-a-default-user-browser
- https://stackoverflow.com/questions/3569749/qt-open-default-file-explorer-on-nix

**Comment by tux3** [comment]
@ovalseven8 
Qt is supposed to use the native file dialog as long as we link dynamically to the system Qt installation, so this is either an upstream problem or a qTox packaging problem.

**Comment by ovalseven8** [comment]
@tux3:
Just compiled qTox on Lubuntu 15.04 (Qt 5.4.1) and, yeah, now the file manager from the system is used. :+1: (Well, still the wrong language, but anyway)
So, the problem is the official packaging.

Hope that in the future the Tox packaging system will improve and then also change to dynamic linking.

**Comment by zetok** [comment]
https://github.com/tux3/qTox/blob/deeb9b2308290e0404198a38a5d4c456ac39b328/INSTALL.md#community-builds ← could you check if it works for you?

**Comment by zetok** [comment]
No response for over a month, assuming it was fixed, thus closing.

**Comment by ovalseven8** [comment]
I have just tested the "new" community builds. Unfortunately, the problem is not solved yet.
Please reopen.

**Comment by zetok** [comment]
cc @abbat ?

**Comment by antis81** [comment]
**EDIT:** After short discussion with @zetok, it appears to be a bug in Qt's native dialog detection - though I cannot actually believe that.

~~@zetok @abbat FYI: A fix will soon be available by @anoadragon453. No need to push that further.~~

**Comment by zetok** [comment]
@antis81 how exactly, since that's a packaging problem?

**Comment by abbat** [comment]
If problem actually with static Qt linking instead dynamic from system Qt, I can not fix it. Debian 7/8 and Ubuntu 12.04 / 14.04 has not required Qt version in repository and we have to link it static.

If we will refactoring qTox code to support Qt 5.2.1 (Ubuntu Trusty) or Qt 5.3.2 (Debian Jessie) we will be able to use system Qt.

**Comment by zetok** [comment]
But with e.g. 16.04 it should work just fine, right?

@ovalseven8 mind testing again with 16.04?

**Comment by ovalseven8** [comment]
Just tested on Lubuntu 16.04:
Both @abbat's qTox and the compiled qTox don't use the file manager from the system. :(

**Comment by zetok** [comment]
@ovalseven8 actually, sorry for misleading a bit – native file picker will be brought back to qTox after #3494 gets fixed, and that's when testing would need to be done.

**Comment by Diadlo** [comment]
Seems, duplicate of #3494, so it will be closed. Please comment if it isn't
